controllers/uploadpackage.lua

```diff
@@ -0,0 +1,108 @@
+local app_helpers = require("lapis.application")
+local validate = require("lapis.validate")
+local capture_errors, assert_error = app_helpers.capture_errors, app_helpers.assert_error
+local lfs = require('lfs')
+local Packages = require("models.packages")
+local Users = require("models.users")
+
+local function isFile(name)
+  if not isDir(name) then
+    return os.rename(name,name) and true or false
+  end
+  return false
+end
+
+local function isFileOrDir(name)
+  if type(name)~="string" then return false end
+  return os.rename(name, name)
+end
+
+local function isDir(name)
+  if type(name)~="string" then return false end
+  local cd = lfs.currentdir()
+  local is, err = lfs.chdir(name)
+  lfs.chdir(cd)
+  return is, err
+end
+
+return {
+  GET = capture_errors(function(self)
+    assert_error(self.session.name, self.i18n("err_not_logged_in"))
+    assert_error(self.session.verified, self.i18n("err_email_not_verified"))
+    return { render = "uploadpackage" }
+  end),
+  POST = capture_errors(function(self)
```

Comment:
One big method - I'd prefer to see it split up into smaller ones that each does it's certain task.

Reply:
I mean, I could break it into "validate parameters" and "Save file" I suppose but that's all it's really doing

Reply:
I broke out the file saving to its own function, which does help it read better.